Tasks/Gradle/gradletask.ts

```diff
@@ -80,8 +106,15 @@ if (javaHomeSelection == 'JDKVersion') {
         var envName = "JAVA_HOME_" + jdkVersion.slice(2) + "_" + jdkArchitecture.toUpperCase();
         specifiedJavaHome = tl.getVariable(envName);
         if (!specifiedJavaHome) {
-            tl.error('Failed to find specified JDK version. Please make sure environment variable ' + envName + ' exists and is set to the location of a corresponding JDK.');
-            tl.exit(1);
+            if (isWindows) {
+                // attempt to discover java home property from registry on Windows
+                specifiedJavaHome = readJavaHomeFromRegistry(jdkVersion, jdkArchitecture);
+            }
+
+            if (!specifiedJavaHome) { 
+                tl.error('Failed to find specified JDK version. Please make sure environment variable ' + envName + ' exists and is set to the location of a corresponding JDK.');
+                tl.exit(1);
```

Comment:
exit is not good and can cause buffered STDOUT to be lost (when you call exit node does not flush first and there is no way to make it flush first - at least i spent hours trying to find a way and no luck). bubble to an outer catch and tl.setresult is a better pattern